packages/flutter/test/widgets/flow_test.dart

```diff
@@ -82,4 +99,23 @@ void main() {
     await tester.tapAt(const Offset(20.0, 90.0));
     expect(log, equals(<int>[0]));
   });
+
+  testWidgets('Flow opacity layer', (WidgetTester tester) async {
+    final double opacity = 0.2;
+    await tester.pumpWidget(
+      new Flow(
+        delegate: new OpacityFlowDelegate(opacity),
+        children: <Widget>[
+          new Container(width: 100.0, height: 100.0),
+        ]
+      )
+    );
+    ContainerLayer layer = RendererBinding.instance.renderView.debugLayer;
+    while (layer != null && !(layer is OpacityLayer))
+      layer = layer.firstChild;
+    expect(layer, const isInstanceOf<OpacityLayer>());
+    final OpacityLayer opacityLayer = layer;
+    expect(opacityLayer.alpha, equals(opacity * 255));
+    expect(layer.firstChild, const isInstanceOf<TransformLayer>());
+  });
```

Comment:
looks good. for completeness' sake, let's also have the opposite test (when opacity == 1.0), just to be sure.

(I assume this test fails without your fix?)

Reply:
The code works the same for all opacity values.  The bug was making the layers affected by opacity siblings of the opacity layer, when they should have been children of the opacity layer.

The malformed tree wasn't noticeable when opacity is 1.0 because the opacity layer would not have affected rendering.